### PR TITLE
OpenID/GitLab challenge options

### PR DESCRIPTION
**Pull request by liggitt**
added in https://github.com/openshift/origin/pull/8732, post 1.2/3.2

### CODE REVIEW

**Comment by liggitt** [comment]
@adellape ping

**Comment by adellape** [comment]
LGTM, labeling for Future/enterprise-3.3 as I don't see the code in 3.2.1.

**Comment by adellape** [comment]
[rev_history]
|link:../install_config/configuring_authentication.html[Configuring Authentication]
|Added link:../install_config/configuring_authentication.html#OpenID[OpenID] and  link:../install_config/configuring_authentication.html#GitLab[GitLab]  challenge options.
%

File: install_config/configuring_authentication.adoc [hunk 1]
```diff
@@ -1004,6 +1004,10 @@ oauthConfig:
 ----
 <1> This provider name is prefixed to the GitLab numeric user ID to form an
 identity name. It is also used to build the callback URL.
+<2> When *true*, unauthenticated token requests from non-web clients (like
+the CLI) are sent a `WWW-Authenticate` challenge header for this provider.
+This uses the http://doc.gitlab.com/ce/api/oauth2.html#resource-owner-password-credentials[Resource Owner Password Credentials]
+grant flow to obtain an access token from GitLab. 
 <2> *GitLabIdentityProvider* cannot be used to send `WWW-Authenticate`
```
Comment (php-coder) [inline install_config/configuring_authentication.adoc]: @liggitt Sorry for bothering you, but I found out that this commits added two callouts with the same number. I mean there was `<2>` already and another one was added. I can provide a fix for this if you told me how (merge, remove previous or something else). Thanks.

Comment (liggitt) [inline install_config/configuring_authentication.adoc]: just open a new PR with the fix

Comment (php-coder) [inline install_config/configuring_authentication.adoc]: Definitely will open but I don't understand how it should be fixed :(

Comment (liggitt) [inline install_config/configuring_authentication.adoc]: open a PR removing the old `<2>` callout line

Comment (php-coder) [inline install_config/configuring_authentication.adoc]: Ok, I added a fix: https://github.com/openshift/openshift-docs/pull/2716/commits/ed5304b8d7d581c5925aef954f7ae558e6ad2b1d

Thank you!

**Pull request merged**